src/Diagnostics.AIProjects/SearchAPI/SearchModule/views.py

```diff
@@ -108,6 +115,28 @@ def queryDetectorsMethod():
     res = json.dumps(results)
     return (res, 200)
 
+@app.route('/queryMultiple', methods=["POST"])
+def queryMultipleMethod():
+    data = json.loads(request.data.decode('utf-8'))
+    requestId = data['requestId']
+
+    txts = data['texts']
+    if not txts:
+        return ("No texts provided for search", 400)
+    productid = getProductId(data)
+    if not productid:
+        return ('Resource data not available', 404)
+    productid = productid[0]
+    try:
+        loadModel(productid)
+    except Exception as e:
+        loggerInstance.logHandledException(requestId, e)
+        loggerInstance.logToFile(requestId, e)
+        return (json.dumps({"query": txts, "results": []}), 200)
```

Comment:
you do want to return 200 in case of exception?

Reply:
Do you want to differentiate at caller side that LoadModel failed vs genuine no results?

Reply:
Yeah we should return a 404 with exception message.